### PR TITLE
FAT-1395/C347874

### PR DESCRIPTION
**Pull request by re-jas**
[Implement e-2-e automation of test case C347874: Overdue fine, returned triggers](https://issues.folio.org/browse/FAT-1395)

### CODE REVIEW

File: cypress/e2e/fee-fine/overdue-fine-returned-triggers.cy.js [hunk 1]
```diff
@@ -0,0 +1,375 @@
...
+          ],
+        }).then((specialInstanceIds) => {
+          itemData.instanceId = specialInstanceIds.instanceId;
+          itemData.holdingId = specialInstanceIds.holdingIds[0].id;
+          itemData.itemId = specialInstanceIds.holdingIds[0].itemIds;
+        });
+      });
+
+    LoanPolicy.createViaApi(loanPolicyBody);
+    OverdueFinePolicy.createApi(overdueFinePolicyBody);
```
Comment (penfriend) [inline cypress/e2e/fee-fine/overdue-fine-returned-triggers.cy.js:219]: OverdueFinePolicy.createViaApi

File: cypress/e2e/fee-fine/overdue-fine-returned-triggers.cy.js [hunk 2]
```diff
@@ -0,0 +1,375 @@
...
+    );
+  });
+
+  it(
+    'C347874 Overdue fine, returned triggers (vega)',
+    { tags: [TestTypes.criticalPath, devTeams.vega] },
+    () => {
+      createPatronNoticeTemplate(noticeTemplates.returnedUponAt);
+      createPatronNoticeTemplate(noticeTemplates.returnedAfterOnce);
+      createPatronNoticeTemplate(noticeTemplates.returnedAfterRecurring);
```
Comment (penfriend) [inline cypress/e2e/fee-fine/overdue-fine-returned-triggers.cy.js:311]: please create this templates using "dublicate" button in the menu "Actions"

File: cypress/e2e/fee-fine/overdue-fine-returned-triggers.cy.js [hunk 3]
```diff
@@ -0,0 +1,375 @@
...
+      NewNoticePolicy.waitLoading();
+      NewNoticePolicy.startAdding();
+      NewNoticePolicy.checkInitialState();
+      NewNoticePolicy.fillGeneralInformation(noticePolicy);
+      NewNoticePolicy.addFeeFineNotice(selectOptions(noticeTemplates.returnedUponAt));
+      NewNoticePolicy.addFeeFineNotice(selectOptions(noticeTemplates.returnedAfterOnce), 1);
+      NewNoticePolicy.addFeeFineNotice(selectOptions(noticeTemplates.returnedAfterRecurring), 2);
+      NewNoticePolicy.save();
+      NewNoticePolicy.waitLoading();
+      NewNoticePolicy.check(noticePolicy);
```
Comment (penfriend) [inline cypress/e2e/fee-fine/overdue-fine-returned-triggers.cy.js:323]: let's rename this method check to "checkPolicyName"

File: cypress/e2e/fee-fine/overdue-fine-returned-triggers.cy.js [hunk 4]
```diff
@@ -0,0 +1,375 @@
...
+        });
+    });
+  });
+
+  after('Deleting created entities', () => {
+    UserEdit.changeServicePointPreferenceViaApi(userData.userId, [testData.userServicePoint.id]);
+    CirculationRules.deleteRuleViaApi(testData.baseRules);
+    ServicePoints.deleteViaApi(testData.userServicePoint.id);
+    cy.deleteLoanPolicy(loanPolicyBody.id);
+    NoticePolicyApi.deleteApi(testData.ruleProps.n);
```
Comment (penfriend) [inline cypress/e2e/fee-fine/overdue-fine-returned-triggers.cy.js:273]: deleteViaApi

File: cypress/e2e/fee-fine/overdue-fine-returned-triggers.cy.js [hunk 5]
```diff
@@ -0,0 +1,375 @@
...
+    OverdueFinePolicy.deleteApi(overdueFinePolicyBody.id);
```
Comment (penfriend) [inline cypress/e2e/fee-fine/overdue-fine-returned-triggers.cy.js:274]: deleteViaApi

File: cypress/e2e/fee-fine/overdue-fine-returned-triggers.cy.js [hunk 6]
```diff
@@ -0,0 +1,375 @@
...
+    Users.deleteViaApi(userData.userId);
+    PatronGroups.deleteViaApi(patronGroup.id);
+    cy.deleteItem(itemData.itemId);
```
Comment (penfriend) [inline cypress/e2e/fee-fine/overdue-fine-returned-triggers.cy.js:277]: deleteViaApi

File: cypress/e2e/fee-fine/overdue-fine-returned-triggers.cy.js (hunk 1 above)
Comment (re-jas) [reply cypress/e2e/fee-fine/overdue-fine-returned-triggers.cy.js:219]: done

File: cypress/e2e/fee-fine/overdue-fine-returned-triggers.cy.js (hunk 3 above)
Comment (re-jas) [reply cypress/e2e/fee-fine/overdue-fine-returned-triggers.cy.js:323]: done

File: cypress/e2e/fee-fine/overdue-fine-returned-triggers.cy.js (hunk 4 above)
Comment (re-jas) [reply cypress/e2e/fee-fine/overdue-fine-returned-triggers.cy.js:273]: done

File: cypress/e2e/fee-fine/overdue-fine-returned-triggers.cy.js (hunk 5 above)
Comment (re-jas) [reply cypress/e2e/fee-fine/overdue-fine-returned-triggers.cy.js:274]: done

File: cypress/e2e/fee-fine/overdue-fine-returned-triggers.cy.js (hunk 6 above)
Comment (re-jas) [reply cypress/e2e/fee-fine/overdue-fine-returned-triggers.cy.js:277]: done

File: cypress/e2e/fee-fine/overdue-fine-returned-triggers.cy.js (hunk 2 above)
Comment (re-jas) [reply cypress/e2e/fee-fine/overdue-fine-returned-triggers.cy.js]: done

**Review by MaksimTerekhov — APPROVED** [review]

File: cypress/support/fragments/circulation/newNoticePolicyTemplate.js [hunk 7]
```diff
@@ -1,6 +1,6 @@
 /* eslint-disable cypress/no-unnecessary-waiting */
 import getRandomPostfix from '../../utils/stringTools';
-import { Button, TextField, TextArea, KeyValue, Checkbox, Link, Heading, Select, Pane, Modal, Section } from '../../../../interactors';
+import { Button, TextField, TextArea, KeyValue, Checkbox, Link, Heading, Select, Pane, Modal, Section, NavListItem } from '../../../../interactors';
```
Comment (penfriend) [inline cypress/support/fragments/circulation/newNoticePolicyTemplate.js:3]: 'NavListItem' is defined but never used. Allowed unused vars must match /React/u

File: cypress/support/fragments/circulation/newNoticePolicyTemplate.js [hunk 8]
```diff
@@ -134,5 +138,20 @@ export default {
...
+
+  duplicateTemplate: () => {
+    cy.do([
+      actionsButton.click(),
+      actionsButtons.duplicate.click(),
+    ]);
+  },
+
+  typeTemplateName: (noticePolicytemplateName) => {
+    cy.get('#input-patron-notice-name').clear().type(noticePolicytemplateName);
```
Comment (penfriend) [inline cypress/support/fragments/circulation/newNoticePolicyTemplate.js:151]: actually we need to use here an interactor TextField, but I am not 100% sure that it works properly in this case

File: cypress/support/fragments/circulation/newNoticePolicyTemplate.js [hunk 9]
```diff
@@ -134,5 +138,20 @@ export default {
...
+  },
+
+  typeTemplateSubject: (noticePolicytemplateSubject) => {
+    cy.get('#input-patron-notice-subject').clear().type(noticePolicytemplateSubject);
```
Comment (penfriend) [inline cypress/support/fragments/circulation/newNoticePolicyTemplate.js:155]: actually we need to use here an interactor TextField, but I am not 100% sure that it works properly in this case

File: cypress/support/fragments/circulation-log/searchPane.js [hunk 10]
```diff
@@ -122,6 +122,10 @@ export default {
     });
   },
 
+  findResultRowIndexByContent(content) {
+    return cy.get('*[class^="mclCell"]').contains(content).parent().invoke('attr', 'data-row-inner');
```
Comment (penfriend) [inline cypress/support/fragments/circulation-log/searchPane.js:126]: can you use here  MultiColumnListRow?

File: cypress/e2e/fee-fine/overdue-fine-returned-triggers.cy.js [hunk 11]
```diff
@@ -0,0 +1,387 @@
...
+      patronGroup.id = res;
+      cy.createTempUser(
+        [
+          permissions.checkinAll.gui,
+          permissions.checkoutAll.gui,
+          permissions.circulationLogAll.gui,
+          permissions.uiCirculationSettingsNoticeTemplates.gui,
+          permissions.uiCirculationSettingsNoticePolicies.gui,
+          permissions.loansAll.gui,
+
```
Comment (penfriend) [inline cypress/e2e/fee-fine/overdue-fine-returned-triggers.cy.js:247]: do you need empty line here?

Comment (re-jas) [reply cypress/e2e/fee-fine/overdue-fine-returned-triggers.cy.js:247]: removed

File: cypress/support/fragments/circulation/newNoticePolicyTemplate.js (hunk 9 above)
Comment (re-jas) [reply cypress/support/fragments/circulation/newNoticePolicyTemplate.js:155]: done

File: cypress/support/fragments/circulation/newNoticePolicyTemplate.js (hunk 8 above)
Comment (re-jas) [reply cypress/support/fragments/circulation/newNoticePolicyTemplate.js:151]: done

File: cypress/support/fragments/circulation/newNoticePolicyTemplate.js (hunk 7 above)
Comment (re-jas) [reply cypress/support/fragments/circulation/newNoticePolicyTemplate.js:3]: removed

**Review by penfriend — APPROVED** [review]

**Comment by penfriend** [comment]
link to the jenkins build: https://jenkins-aws.indexdata.com/job/Testing/job/Cypress%20tests/698/allure/

**Pull request merged**